XamlAnimatedGif.Shared/Animator.cs

```diff
@@ -553,6 +561,46 @@ private static Task<Stream> GetStreamFromUriAsync(Uri uri)
             }
             throw new NotSupportedException("Only ms-appx:, ms-appdata:, ms-resource: and file: URIs are supported");
         }
+
+        private static async Task<Stream> GetNetworkStreamAsync(Uri uri)
+        {
+            //generating temp file name by converting the uri to base64
+            var tempId = Convert.ToBase64String(Encoding.UTF8.GetBytes(uri.AbsoluteUri));
+
+            var folder = ApplicationData.Current.TemporaryFolder;
+
+            try
+            {
+                var tempFile = await folder.GetFileAsync(tempId);
+
+                //return the cache image
+                return await tempFile.OpenStreamForReadAsync();
+            }
+            catch{}
```

Comment:
Catch `FileNotFoundException` here; any other exception is unexpected and shouldn't be caught.